src/linear_algebra/affine_space/combination.lean

```diff
@@ -296,6 +296,72 @@ end
 
 end finset
 
+namespace finset
+
+variables (k : Type*) {V : Type*} {P : Type*} [division_ring k] [add_comm_group V] [module k V]
+variables [affine_space V P] {ι : Type*} (s : finset ι)
+
+/-- The weights for the centroid of some points. -/
+@[nolint unused_arguments]
+def centroid_weights (i : ι) := (finset.card s : k) ⁻¹
+
+/-- `centroid_weights` at any point. -/
+@[simp] lemma centroid_weights_apply (i : ι) : s.centroid_weights k i = (finset.card s : k) ⁻¹ :=
+rfl
+
+/-- `centroid_weights` equals a constant function. -/
+lemma centroid_weights_eq_const :
+  s.centroid_weights k = function.const ι ((finset.card s : k) ⁻¹) :=
+rfl
+
+variables {k}
+
+/-- The weights in the centroid sum to 1, if the number of points,
+converted to `k`, is not zero. -/
+lemma sum_centroid_weights_eq_one_of_cast_card_ne_zero (h : (card s : k) ≠ 0) :
+  ∑ i in s, s.centroid_weights k i = 1 :=
+by simp [h]
+
+variables (k)
+
+/-- In the characteristic zero case, the weights in the centroid sum
+to 1 if the number of points is not zero. -/
+lemma sum_centroid_weights_eq_one_of_card_ne_zero [char_zero k] (h : card s ≠ 0) :
+  ∑ i in s, s.centroid_weights k i = 1 :=
+by simp [h]
+
+/-- In the characteristic zero case, the weights in the centroid sum
+to 1 if the set is nonempty. -/
+lemma sum_centroid_weights_eq_one_of_nonempty [char_zero k] (h : s.nonempty) :
+  ∑ i in s, s.centroid_weights k i = 1 :=
+s.sum_centroid_weights_eq_one_of_card_ne_zero k (ne_of_gt (card_pos.2 h))
+
+/-- In the characteristic zero case, the weights in the centroid sum
+to 1 if the number of points is `n + 1`. -/
+lemma sum_centroid_weights_eq_one_of_card_eq_add_one [char_zero k] {n : ℕ}
+  (h : card s = n + 1) : ∑ i in s, s.centroid_weights k i = 1 :=
+s.sum_centroid_weights_eq_one_of_card_ne_zero k (h.symm ▸ nat.succ_ne_zero n)
+
+include V
```

Comment:
Is this necessary? If this is a consequence of Yury'd recent work on implicit arguments then it may be worth an implementation note (apologies if it's already there, I'm reviewing this quickly before going to bed).

Reply:
Yes, it's needed as a result of the `out_param` work (the linter will tell you if any individual definitions need `omit V` because they aren't using `V` at all). Added a note in `linear_algebra/affine_space/basic.lean`.